IPython/core/completer.py

```diff
@@ -998,6 +992,8 @@ def _make_signature(completion)-> str:
 class IPCompleter(Completer):
     """Extension of the completer class with IPython-specific features"""
 
+    _names = None
```

Comment:
When is '_names' changes?

Reply:
This is a tiny bit complex, but here we have *class variable*, in some case it is equivalent to `self._name`. In this particular case, that would be equivalent to 

```
def __init__(self, *args, **kwargs):
    self._names = None
    ...
````